### PR TITLE
refactor: feature flag typing remove key-value

### PR DESCRIPTION
**Pull request by stephanwlee**
Key-value allows just about anything and it is not that great for
typing. This change removes the key-value part of the FeatureFlag and
forces all apps to strongly type it. We can reintroduce it when we
absolutely need it.

### CODE REVIEW

File: tensorboard/webapp/feature_flag/effects/feature_flag_effects_test.ts [hunk 1]
```diff
@@ -62,6 +62,7 @@ describe('feature_flag_effects', () => {
     it('loads features from the data source on init', () => {
       spyOn(dataSource, 'getFeatures').and.returnValue({
         enabledExperimentalPlugins: ['foo', 'bar'],
+        inColab: false,
```
Comment (psybuzz) [inline tensorboard/webapp/feature_flag/effects/feature_flag_effects_test.ts:65]: Perhaps we can use `buildFeatureFlags({enabledExperimentalPlugins: ['foo', 'bar']})` and drop the `inColab` here and below?

**Review by psybuzz — COMMENTED** [review]
Thanks for making typing safer :)

I sprinkled a few comments with the purpose of reducing object literals
that require 'all feature flags as keys' into as few files as possible.
It would be great if, when we need to introduce 1 more feature flag,
we would only have to touch `feature_flag/types.ts` and no other file
under `feature_flag/`.

File: tensorboard/webapp/feature_flag/testing.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+import {FeatureFlags} from './types';
+
+export function buildFeatureFlag(
```
[the remark this answers is not in the document]
Comment (stephanwlee) [reply tensorboard/webapp/feature_flag/testing.ts:18]: I want types to be exclusively about the types, not for test utils.

File: tensorboard/webapp/feature_flag/effects/feature_flag_effects_test.ts (hunk 1 above)
Comment (stephanwlee) [reply tensorboard/webapp/feature_flag/effects/feature_flag_effects_test.ts:65]: I will not remove it.

I mean, I am not going to pass every single possible thing but this test spec is neither specifically about the `enabledExperimentalPlugins` or `inColab`. However, exercising both is a valid test imo.

File: tensorboard/webapp/feature_flag/testing.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
```
Comment (psybuzz) [inline tensorboard/webapp/feature_flag/testing.ts:1]: nit: 2020

File: tensorboard/webapp/feature_flag/effects/feature_flag_effects_test.ts (hunk 1 above)
Comment (psybuzz) [reply tensorboard/webapp/feature_flag/effects/feature_flag_effects_test.ts:65]: This list of files that require all feature flags (selectors.ts, effects_test.ts, testing.ts, types.ts) is small enough that the PR as-is looks reasonable to me.

**Review by psybuzz — APPROVED** [review]

**Pull request merged**